### PR TITLE
Generic pipelines

### PR DESCRIPTION
**Pull request by lbesnard**

### CODE REVIEW

**Comment by ggalibert** [comment]
Same with commit https://github.com/aodn/python-aodndata/pull/20/commits/b824125012260aee324e819312976829d0207cc5 which should only include changes relevant to CARS. There seemed to be a mixed up there.

**Comment by lbesnard** [comment]
hum i know what happened. A mix of pycharm adding files to git, and me doing it via the terminal...

**Comment by lbesnard** [comment]
@ggalibert done

**Comment by ggalibert** [comment]
Will leave it to @lwgordonimos to finally approve. Looks like Travis is still complaining...

**Comment by lbesnard** [comment]
Travis is happy !

File: aodndata/csiro/cars.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+def dest_path_cars(filepath):
+    try:
+        pattern = 'CARS(\d+)_.*\.nc'
+        year = re.search(pattern, filepath).group(1)
+    except AttributeError:
+        raise InvalidFileNameError(
+            'invalid file name {filepath}. Not matching \'CARS(\d+)_.*\.nc\''.format(filepath=filepath))
+    return os.path.join('CSIRO/Climatology/CARS/%s/AODN-product/%s' % (year, os.path.basename(filepath)))
```
Comment (lwgordonimos) [inline aodndata/csiro/cars.py]: str.format is used elsewhere, so would be good to keep consistent, e.g.

```python
return os.path.join("CSIRO/Climatology/CARS/{year}/AODN-product/{basename}".format(year=year, basename=os.path.basename(filepath))
```

File: aodndata/csiro/ocean_acidification.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import os
+
+def dest_path_oa(filepath):
+    return 'CSIRO/Climatology/Ocean_Acidification/%s' % os.path.basename(filepath)
```
Comment (lwgordonimos) [inline aodndata/csiro/ocean_acidification.py]: Same re str.format

File: test_aodndata/csiro/test_cars.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import unittest
+
+from aodndata.csiro.cars import dest_path_cars
+from aodncore.pipeline.exceptions import InvalidFileNameError
+
+
+class TestCarsHandler(unittest.TestCase):
+
+    def test_dest_path_cars(self):
+        good_nc = 'CARS2009_World_monthly_01.nc'
```
Comment (lwgordonimos) [inline test_aodndata/csiro/test_cars.py]: Add directory structure here to test basename stuff

**Pull request merged**